tls/s2n_connection.h

```diff
@@ -85,6 +101,11 @@ struct s2n_connection {
     uint8_t actual_protocol_version;
     uint8_t actual_protocol_version_established;
 
+    /* Certificate Authentication and Verification Parameters */
+    s2n_cert_auth_type client_cert_auth_type;
```

Comment:
Could we name this "cert_auth_type" or "peer_auth_type" and remove the "client" prefix? Everything else you've added in the struct looks nicely generalized to suit both the server auth and the client auth side. If s2n_mode is S2N_SERVER, then this could default to NONE, and if s2n_mode is S2N_CLIENT, then this could default to REQUIRED rather than being included and unused for the S2N_CLIENT case.